metis-common/src/main/java/eu/europeana/metis/mongo/MongoClientProvider.java

```diff
@@ -69,23 +78,35 @@ public MongoClientProvider(String connectionUri, Function<String, E> exceptionCr
    *
    * @param properties The properties of the Mongo connection. Note that if the passed properties
    * object is changed after calling this method, those changes will not be reflected when calling
-   * @param optionsBuilder The settings to be applied. The default settings will not be used. The
-   * caller can incorporate the default settings by using an options builder obtained from {@link
-   * #getDefaultOptionsBuilder()}. {@link #createMongoClient()}.
+   * @param clientSettingsBuilder The settings to be applied. The default settings will not be used. The
+   * caller can incorporate the default settings by using an client settings builder obtained from {@link
+   * #getDefaultClientSettingsBuilder()}. {@link #createMongoClient()}.
    */
-  public MongoClientProvider(MongoProperties<E> properties, Builder optionsBuilder) {
+  public MongoClientProvider(MongoProperties<E> properties, Builder clientSettingsBuilder) {
     final ReadPreference readPreference = Optional.ofNullable(properties.getReadPreferenceValue())
-            .map(ReadPreferenceValue::getReadPreferenceSupplier).map(Supplier::get)
-            .orElse(DEFAULT_READ_PREFERENCE);
-    optionsBuilder.sslEnabled(properties.mongoEnableSsl()).readPreference(readPreference);
+        .map(ReadPreferenceValue::getReadPreferenceSupplier).map(Supplier::get)
+        .orElse(DEFAULT_READ_PREFERENCE);
+    clientSettingsBuilder.readPreference(readPreference);
+
     final MongoCredential mongoCredential = properties.getMongoCredentials();
     this.authenticationDatabase = Optional.ofNullable(mongoCredential)
-            .map(MongoCredential::getSource).orElse(null);
+        .map(MongoCredential::getSource).orElse(null);
+    clientSettingsBuilder.applyToSslSettings(builder -> builder.enabled(properties.mongoEnableSsl()));
     if (mongoCredential == null) {
-      this.creator = () -> new MongoClient(properties.getMongoHosts(), optionsBuilder.build());
+      this.creator = () -> {
+        final List<ServerAddress> mongoHosts = properties.getMongoHosts();
+        final MongoClientSettings mongoClientSettings = clientSettingsBuilder
+            .applyToClusterSettings(builder -> builder.hosts(mongoHosts)).build();
```

Comment:
True but the problem is that the properties.getMongoHosts(); throws an exception, which was controlled and thrown by the createMongoClient methods, so I was trying to avoid the class signature.
But since you feel it should be changed I updated the signature and the constructor throws an exception as well now.